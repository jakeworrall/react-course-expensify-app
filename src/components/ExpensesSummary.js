import React from "react";
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
    <div>
        Viewing {expenseCount} {expenseCount === 1 ? 'expense':'expenses'} totalling {
        numeral(expensesTotal/100).format('$0,0.00')}
    </div>
);

const mapStateToProps = (state) => ({
    expenseCount: state.expenses.length,
    expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpensesSummary);