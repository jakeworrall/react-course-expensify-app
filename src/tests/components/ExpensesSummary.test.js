import {ExpensesSummary} from "../../components/ExpensesSummary";
import {shallow} from "enzyme/build";
import React from "react";

test('should display summary from 1 expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesTotal={109500} expenseCount={1}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should display summary from multiple expense correctly', () => {
    const wrapper = shallow(<ExpensesSummary expensesTotal={114195} expenseCount={3}/>);
    expect(wrapper).toMatchSnapshot();
});