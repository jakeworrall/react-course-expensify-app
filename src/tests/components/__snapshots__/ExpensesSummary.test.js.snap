// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display summary from 1 expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $1,095.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should display summary from multiple expense correctly 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       
      expenses
       totalling 
      <span>
        $1,141.95
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
